### PR TITLE
Improve docs on DataTables (Fixes #459)

### LINKED ISSUE #459: Datatable could use some improvements
https://cucumber.io/docs/cucumber/api/#step-arguments

While this page explains that tables can be converted to lists of strings it doesn't explain that other types are also possible.

At the same time this page explains how data table types can be configured but it doesn't explain what they do:

https://cucumber.io/docs/cucumber/configuration/

It also looks like the syntax highlighting broke with the migration to smartbear.

### PR DESCRIPTION
**Pull request by mlvandijk**
Add a little more information, as well as links to other documentation (to prevent duplication).

### CODE REVIEW

File: content/docs/cucumber/api.md [hunk 1]
```diff
@@ -23,7 +23,23 @@ The number of parameters in the {{% stepdef-body %}} has to match the number of
 
 ## Data Tables
 
-The simplest way to pass a {{% text "java,kotlin" %}}`List<String>`{{% /text %}}{{% text "scala" %}}`java.util.List[String]`{{% /text %}} to a step definition is to use a data table:
+Data tables from Gherkin can be accessed by using the DataTable object as the last parameter in a step definition.
+This conversion can be done either by Cucumber or manually.
+
+{{% text "java,kotlin" %}}Depending on the table shape as one of the following collections:{{% text "java,kotlin" %}}
```
Comment (olleolleolle) [inline content/docs/cucumber/api.md]: This {{ seems unpaired - the opening and closing {{'s do not line up, do they?

Comment (mpkorstanje) [reply content/docs/cucumber/api.md:29]: Looks perfectly balanced to me.

Comment (mlvandijk) [reply content/docs/cucumber/api.md]: You are correct. Thanks for catching this.

Comment (mlvandijk) [reply content/docs/cucumber/api.md]: Should be fixed now

File: content/docs/cucumber/api.md [hunk 2]
```diff
@@ -57,9 +73,12 @@ Given("the following animals:") { animals: java.util.List[String] =>
 ```
 {{% /text %}}
 
-In this case, the `DataTable` is automatically flattened to a {{% text "java,kotlin" %}}`List<String>`{{% /text %}}{{% text "scala" %}}`java.util.List[String]`{{% /text %}}
+In this case, the `DataTable` is automatically flattened to a {{% text "java,kotlin" %}}`List<String>`{{% /text %}}{{% text "scala" %}}`java.util.List[String]`{{% /text %}}{{% text "ruby,javascript" %}}list of strings{{% /text %}}
```
Comment (olleolleolle) [inline content/docs/cucumber/api.md]: Minor: If one were a stickler for details, one _could_ call the Ruby & JS list an _array._

File: content/docs/cucumber/api.md [hunk 3]
```diff
@@ -57,9 +73,12 @@ Given("the following animals:") { animals: java.util.List[String] =>
...
 by Cucumber (using `DataTable.asList(String.class)`) before invoking the step definition.
 
+{{% text "java,kotlin" %}}Note: In addition to collections of String, Integer, Float, BigInteger and BigDecimal, Byte,
```
Comment (olleolleolle) [inline content/docs/cucumber/api.md:79]: Minor: What is also supported, is that _lists_ of those data types, _or_ those data types themselves?

**Review by olleolleolle — APPROVED** [review]
Thanks for the improved docs!

File: content/docs/cucumber/api.md (hunk 3 above)
Comment (mlvandijk) [reply content/docs/cucumber/api.md:79]: Collections include lists

File: content/docs/cucumber/api.md (hunk 2 above)
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
In this case, the `DataTable` is automatically flattened to a {{% text "java,kotlin" %}}`List<String>`{{% /text %}}{{% text "scala" %}}`java.util.List[String]`{{% /text %}}{{% text "ruby,javascript" %}}array of strings{{% /text %}}
```

Comment (mlvandijk) [reply content/docs/cucumber/api.md]: Updated, thnx

**Pull request merged**